lambdas/process-journey-event/src/main/java/uk/gov/di/ipv/core/processjourneyevent/ProcessJourneyEventHandler.java

```diff
@@ -413,4 +417,33 @@ private void sendSubJourneyStartAuditEvent(
                         new AuditExtensionSubjourneyType(journeyType),
                         new AuditRestrictedDeviceInformation(deviceInformation)));
     }
+
+    private boolean isPageState(JourneyState journeyState)
+            throws StateMachineNotFoundException, UnknownStateException {
+        StateMachine stateMachine = stateMachines.get(journeyState.subJourney());
+        if (stateMachine == null) {
+            throw new StateMachineNotFoundException(
+                    String.format(
+                            "State machine not found for journey type: '%s'",
+                            journeyState.subJourney()));
+        }
+        return stateMachine.isPageState(journeyState);
+    }
+
+    private BasicState journeyStateToBasicState(JourneyState journeyState) {
+        return (BasicState)
+                stateMachines.get(journeyState.subJourney()).getState(journeyState.state());
+    }
+
+    private JourneyState journeyStateFrom(JourneyChangeState journeyChangeState) {
+        return new JourneyState(
+                journeyChangeState.getJourneyType(), journeyChangeState.getInitialState());
+    }
+
+    private boolean isBackEventDefinedOnState(JourneyState journeyState) {
+        return ((BasicState)
+                        stateMachines.get(journeyState.subJourney()).getState(journeyState.state()))
+                .getEvents()
+                .containsKey(BACK_EVENT);
+    }
```

Comment:
Yeah I agree - I wanted to push as much of this into the state machine, but it's hard hecause of the way our collection of state machines is structured. It would perhaps be better to just have on state machine that knows about all the journey maps.

Reply:
Yep, agreed - the distinction is really for ease of definition. Optimising the execution isn't really that important (not until we get _really_ complicated!)